### DISCUSSION TITLE
DHCP shared networks

### DISCUSSION THREAD

**Discussion by moffitj** [netbox-community/netbox#5793] [Q&A]
I'm still evaluating if netbox would be a good fit for us.

The IPAM looks well featured, but I'm not seeing any support for shared networks. For example, in our DHCP server we might have a /25 and a /27 sharing the same VLAN. If one of them runs out of available leases then the other could still be used for all devices on that VLAN, and is normal and expected.

Does Netbox have any way to handle shared-subnets? Ideally they would show up the way that Prefixes do with the Utilization bar (so that when populated with information exported from the DHCP server we have a good look at lease availability). Conceptually they would be a parent of the prefix item, with each shared-network having one or more prefixes.

Is this possible natively or with a plugin?

**Comment by jhammond-git** [comment]
Not natively that I know of, no.  If it were to be done, I would think that would be something that could be added to the vlan view (/ipam/vlans/####) fairly easily.  I'm not sure if that's something that would be accepted as a feature request, but it wouldn't hurt to try.

**Reply by jhammond-git** [reply]
I should clarify - the ability to show utilization on the prefixes assigned to a vlan in the vlan view might be a something worth asking for.

**Reply by candlerb** [reply]
> I should clarify - the ability to show utilization on the prefixes assigned to a vlan in the vlan view might be a something worth asking for.

I think what you should do is to add a Custom Link to the VLAN and/or Prefix pages, which queries your DHCP server for the number of active leases in the relevant prefix(es) and displays the % utilisation.

Note that Netbox currently doesn't show DHCP "usage" at all.  If you allocate a DHCP pool by creating IP addresses of state "DHCP", that just means "these addresses are available for DHCP server to allocate".  If you fill the entire block with these, Netbox will show the the block as 100% allocated (which is true), regardless of how many leases are active.

You are perhaps thinking that you could create an address of state "DHCP" only when the DHCP server allocates it, and remove it again when the lease expires.  Once ranges (#834) are added, this may become more tempting.  However, that's not how it's intended to be used. 

Netbox isn't supposed to be a portal on the active state of the network, and _"automated import of live network state is strongly discouraged"_[^1]; it's supposed to be a curated source-of-truth as to how the network *should* be, which can in turn be used to populate other systems. In other words, Netbox should be pushing data to your DHCP server, not vice versa.

It is free software, and of course you can use it how you like - but a feature request which goes against this core philosophy is likely to be rejected.

[^1] Reference: https://netbox.readthedocs.io/en/stable/#serve-as-a-source-of-truth

**Comment by candlerb** [comment]
You can create two Prefixes, and associate them with the same VLAN.  This models multiple distinct prefixes on the same layer 2 domain, which I think is what you mean by shared-subnet.  As they are different prefixes they have distinct gateway addresses, so the router needs multiple address aliases (i.e. "secondary" addresses in Cisco-speak)

Normally though, a DHCP range is not an entire prefix.  e.g. for the subnet 192.168.1.0/24, you'd have to reserve at least 192.168.1.1 for the router, so the pool would be no more than 192.168.1.2 - 192.168.1.254.  There's a feature request for "ranges" (#834) in Netbox but that hasn't been implemented yet, so for now what you do is create separate IP address objects for 192.168.1.2 to 192.168.1.254 inclusive, with status "DHCP".
